vmware-tanzu/tanzu-framework
# Add core.tanzu.vmware.com/v1alpha2 Capability resource

### What this PR does / why we need it

API group needs bumping because of the new incoming FeatureGate API changes, which means Capability resource should be at v1alpha2 as well.

This commit does the following:

* Adds new v1alpha2 Capability types which are identical to v1alpha1 types.
* Updates CRD manifest with the new version. The v1alpha1 version is no longer served and v1alpha2 is set as the storage version. Since the types are identical and we don't serve v1alpha1 anymore, we don't need a conversion webhook. The controller just reads all existing v1alpha1 resources and writes it back in the new storage version v1alpha2.
* Changes capability controller to work on v1alpha2 types.
* Updates docs.

### Which issue(s) this PR fixes


Fixes #3835 

### Describe testing done for PR



1. Install old v1alpha1 Capability CRD.
2. Run the capabilities controller and create a v1alpha1 CR:
```yaml
apiVersion: core.tanzu.vmware.com/v1alpha1
kind: Capability
metadata:
  name: tkg-capabilities
  namespace: tkg-system
spec:
  serviceAccountName: "tanzu-capabilities-manager-default-sa"
  queries:
    - name: "tanzu-cluster-with-capability"
      groupVersionResources:
        - name: "capability-resource"
          group: "core.tanzu.vmware.com"
          versions:
            - v1alpha1
          resource: "capabilities"
```
3. Update the Capability API to v1alpha2 (v1alpha1 is no longer stored) and re-run the controller.
4. Controller should re-write all existing CRs to v1alpha2.
```yaml
apiVersion: core.tanzu.vmware.com/v1alpha2
kind: Capability
metadata:
  name: tkg-capabilities
  namespace: tkg-system
spec:
  serviceAccountName: "tanzu-capabilities-manager-default-sa"
  queries:
    - name: "tanzu-cluster-with-capability"
      groupVersionResources:
        - name: "capability-resource"
          group: "core.tanzu.vmware.com"
          versions:
            - v1alpha1
          resource: "capabilities"
```
### Release note

```release-note
`capabilities.core.tanzu.vmware.com` has been updated to v1alpha2 and is identical to v1alpha1. The v1alpha1 Capability API is no longer served. Clients must immediately update to using the v1alpha2 API.
```



### Additional information

#### Special notes for your reviewer
